stream-chat-android-core/src/main/java/io/getstream/chat/android/client/call/ReturnOnErrorCall.kt

```diff
@@ -35,27 +38,42 @@ public class ReturnOnErrorCall<T : Any>(
 ) : Call<T> {
 
     private var job: Job? = null
+    private val canceled = AtomicBoolean(false)
 
-    override fun execute(): Result<T> = runBlocking {
-        originalCall.execute().let {
-            if (it.isSuccess) it
-            else onErrorReturn(it.error())
-        }
-    }
+    override fun execute(): Result<T> = runBlocking { await() }
 
     override fun enqueue(callback: Call.Callback<T>) {
-        originalCall.enqueue { originalResult ->
-            if (originalResult.isSuccess) callback.onResult(originalResult)
-            else job = scope.launch {
-                val result = onErrorReturn(originalResult.error())
-                withContext(DispatcherProvider.Main) {
-                    callback.onResult(result)
+        scope.launch {
```

Comment:
It seems like there is no way to cancel the first `launch`.

Reply:
Cancel the first launch will do nothing, because inside of this coroutine we only enqueue the inner call.
BTW, the `originalCall` is cancelled, and when `enqueue()` is used, no `callback.onResult()` is invoked, so second job won't be created